lib/src/models/comment_impl.dart

```diff
@@ -351,6 +355,35 @@ class Comment extends CommentRef
     setData(this, data);
   }
 
+  @override
+  Future refresh() async {
+    final path = submission.infoPath + '_/' + _id;
+    final params = {
+      'context': '100',
+    };
+    // TODO(bkonyi): clean-up this so it's objectified in a nicer way?
+    final commentList = (await reddit.get(path, params: params))[1]['listing'];
+    final queue = new Queue.from(commentList);
+    var comment;
+    while (queue.isNotEmpty && ((comment == null) || (comment._id != _id))) {
```

Comment:
Just so it's clear what's going on here, the response we get from Reddit may not have the comment we're refreshing at the root of the comment forest it returns, so we need to find it. This just iterates over the tree until we find the comment we're looking for.

Reply:
Thanks.